webapp/src/components/AssetPage/SaleActionBox/BuyNFTButtons/BuyNFTButtons.tsx

```diff
@@ -62,10 +76,29 @@ const BuyNFTButtons = ({
           if (asset && shouldOpenBuyWithCryptoModal) {
             onBuyWithCrypto(asset, order)
           }
+          const isItemFree = !isNFT(asset) && asset.price === '0'
+          const isBuyingEntirelyWithCredits =
+            useCredits &&
+            ((!isNFT(asset) && BigInt(credits.totalCredits) >= BigInt(asset.price)) ||
+              (isNFT(asset) && order && BigInt(credits.totalCredits) >= BigInt(order.price)))
```

Comment:
Would that work as well?
```suggestion
            !isNFT(asset) ? BigInt(credits.totalCredits) >= BigInt(asset.price)) : order && BigInt(credits.totalCredits) >= BigInt(order.price)))
```